src/app/callbacks.js

```diff
@@ -224,28 +217,37 @@ createNameSpace('realityEditor.app.callbacks');
      * Callback for realityEditor.app.getMatrixStream
      * Gets triggered ~60FPS when the AR SDK sends us a new set of modelView matrices for currently visible objects
      * Stores those matrices in the draw module to be rendered in the next draw frame
-     * @param {Object.<string, Array.<number>>} visibleObjects
+     * @param {Object.<string, Array.<number>>} visibleTargets
      */
-    function receiveMatricesFromAR(visibleObjects) {
+    function receiveMatricesFromAR(visibleTargets) {
         if (!realityEditor.worldObjects) {
             return;
         } // prevents tons of error messages while app is loading but Vuforia has started
         
         // If viewing the VR map instead of the AR view, don't update objects/tools based on Vuforia
         if (!realityEditor.device.modeTransition.isARMode()) return;
 
-        // this first section makes the app work with extended or non-extended tracking while being backwards compatible
-
-        if (!matrixFormatCalculated) {
-            // for speed, only calculates this one time
-            calculateMatrixFormat(visibleObjects);
-        }
-
-        // ignore this step if using old app version that ignores EXTENDED_TRACKED objects entirely
-        if (isMatrixFormatNew) {
-            // extract status into separate data structure and and format matrices into a backwards-compatible object
-            // if extended tracking is turned off, discard EXTENDED_TRACKED objects
-            convertNewMatrixFormatToOld(visibleObjects);
+        // in older versions, visibleTargets maps objectIds directly to a matrix
+        // in newer versions, visibleTargets maps targetIds to { matrix: [], targetName: "" }
+        // resolves incompatible app/server/object versions by finding the object that matches either the objectId,
+        // targetName, or whose targetId matches the targetId
+        let visibleObjects = {};
+        for (let key in visibleTargets) {
+            if (Array.isArray(visibleTargets[key])) {
+                // backwards compatible with old app versions -> the key is the objectId, and it directly holds the matrix
+                visibleObjects[key] = visibleTargets[key];
+            } else {
+                // find the object whose targetId matches the key, and store the .matrix under the objectId key
+                let matchingObjectId = Object.keys(objects).find(objectKey => {
+                    return objects[objectKey].targetId === key;
+                });
+                if (matchingObjectId) {
+                    visibleObjects[matchingObjectId] = visibleTargets[key].matrix;
+                } else {
+                    // for old objects that weren't generated with a targetId, try to find them by targetName
+                    visibleObjects[visibleTargets[key].targetName] = visibleTargets[key].matrix;
```

Comment:
So status and status info of targets was never propagated further into the JS side of system ?

Reply:
Currently there was nothing making use of the status / status info, so we just passed the matrix for simplicity/efficiency. If you can think of a way to make use of that data then I can change it to include the status / status info.

Reply:
I was thinking about notifications about potentially inaccurate camera tracking. This should be a combination of status from device tracking and area target tracking (coming through visibleTargets here). But that's for another PR.

Reply:
That makes sense! We do have some of this functionality, as the status and statusInfo are included in the getCameraMatrixStream callback https://github.com/ptcrealitylab/vuforia-spatial-toolbox-userinterface/blob/70a35bd58659c4186a75f28effe69ae6bd78cd77/src/app/callbacks.js#L425 and handled in the src/device/tracking.js https://github.com/ptcrealitylab/vuforia-spatial-toolbox-userinterface/blob/70a35bd58659c4186a75f28effe69ae6bd78cd77/src/device/tracking.js#L91

So the device tracking side of things is already accounted for, but I agree it would be good to incorporate the status from the area target tracking, too.